state/protocol/protocol_state.go

```diff
@@ -177,3 +179,37 @@ type StateMutator interface {
 	//     of this function is not nil. If such an exception is returned, continuing is not an option.
 	ApplyServiceEventsFromValidatedSeals(seals []*flow.Seal) error
 }
+
+type DeferredDBUpdate func(tx *transaction.Tx, blockID flow.Identifier) error
```

Comment:
I would suggest to call this struct `DeferredBlockPersistOp`, because we are literally letting the state machines specify the db operations to persist their respective the sub-state. All that persisted data is strictly tied to a block, which I think is important to reflect in the code.